x-pack/plugins/spaces/server/lib/request_interceptors/on_post_auth_interceptor.ts

```diff
@@ -106,7 +106,10 @@ export function initSpacesOnPostAuthRequestInterceptor({
         }
       }
 
-      const allFeatures = features.getKibanaFeatures();
+      // The spaces client returns migrated feature IDs in `disabledFeatures`, so we need to omit deprecated features.
+      // Otherwise apps granted by deprecated features will be considered available when they shouldn't be,
+      // since their IDs won't be present in `disabledFeatures`.
+      const allFeatures = features.getKibanaFeatures({ omitDeprecated: true });
```

Comment:
Let me take a closer look at this specific part tomorrow.

Reply:
Okay, I double-checked, and I agree that omitting deprecated features is the correct approach since we should deal only with non-deprecated features here, and only the apps provided by such features are relevant (not specific to the user or their roles). However, as I mentioned in the x-pack/plugins/features/server/plugin.ts file, we shouldn’t extend deprecated methods to achieve this. Instead, we should expose an `omitDeprecated` option from the `getFeatures` method on the `start` contract.

That said, this issue isn’t caused by your PR - it’s something we need to address on the platform side. Let me try to take a stab at this next week, and I’ll either push changes to your PR or create a dedicated one for it. If/when you’re close to merging this PR and haven’t heard from me by then, please DM me. Thanks!